lib/improver/blending/weighted_blend.py

```diff
@@ -811,25 +689,113 @@ def weighted_mean(self, cube, weights):
+        else:
+            frt_units = frt_coord.units.origin
+            frt_calendar = frt_coord.units.calendar
+            cycletime_point = cycletime_to_number(
+                    cycletime, time_unit=frt_units, calendar=frt_calendar)
+            return np.round(cycletime_point).astype(np.int64)
+
+    def _set_coords_to_remove(self, input_cube):
+        """
+        Generate a list of coordinate names associated with the blend
+        dimension.  Unless these are time-related coordinates, they should be
+        removed after blending.
+
+        Args:
+            input_cube (iris.cube.Cube):
+                Cube to be blended
+        """
+        time_coords = ["time", "forecast_reference_time", "forecast_period"]
+        blend_dim, = input_cube.coord_dims(self.blend_coord)
+        self.crds_to_remove = []
+        for coord in input_cube.coords():
+            if coord.name() in time_coords:
+                continue
+            if blend_dim in input_cube.coord_dims(coord):
+                self.crds_to_remove.append(coord.name())
+
+    def _set_forecast_reference_time_and_period(self, blended_cube):
+        """
+        For cycle and model blending, update the forecast reference time and
+        forecast period coordinate points to the single most recent value,
+        rather than the blended average, and remove any bounds from the
+        forecast reference time. Modifies cube in place.
+
+        Args:
+            blended_cube (iris.cube.Cube)
+        """
+        blended_cube.coord("forecast_reference_time").points = [
+            self.cycletime_point]
+        blended_cube.coord("forecast_reference_time").bounds = None
+        if blended_cube.coords("forecast_period"):
+            blended_cube.remove_coord("forecast_period")
+        new_forecast_period = forecast_period_coord(blended_cube)
+        time_dim = blended_cube.coord_dims("time")
+        blended_cube.add_aux_coord(new_forecast_period, data_dims=time_dim)
+
+    def _update_blended_metadata(self, blended_cube, attributes_dict):
+        """
+        Update metadata after blending:
+        - For cycle and model blending, set a single forecast reference time
+        and period using self.cycletime_point or the latest cube contributing
+        to the blend
+        - Remove scalar coordinates that were previously associated with the
+        blend dimension
+        - Update attributes as specified via process arguments
+        Modifies cube in place.
+
+        Args:
+            blended_cube (iris.cube.Cube)
+            attributes_dict (dict or None)
+        """
+        if self.blend_coord in ["forecast_reference_time", "model_id"]:
+            self._set_forecast_reference_time_and_period(blended_cube)
+        for coord in self.crds_to_remove:
+            blended_cube.remove_coord(coord)
+        if attributes_dict is not None:
+            amend_attributes(blended_cube, attributes_dict)
+
+    def process(self, cube, weights=None,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameters differ from overridden 'process' method](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4464994)

Reply:
@benfitzpatrick `BasePlugin.process` has `(self, *args, **kwargs)` - can we get Codacy to be quiet?

Reply:
@benfitzpatrick we think Codacy might have to get used to this since #910 !